### DISCUSSION TITLE
unable to read data from oracle catalog via trino

### DISCUSSION THREAD

**Discussion by vmsaipreeth** [ibis-project/ibis#6542] [Q&A] [answered]
We are using trino as our backend. 
```
connection = ibis.trino.connect( trino_connection_parameters)

df_ibis = connection.sql(“””SQL STATEMENT WITH ORACLE CATALOG VIA TRINO”””)
```

This throws an error: 
```
ParseError: expected 'EOF' at 0:7
```

Note: This works fine for hdfs and object storage catalogs

**Comment by gforsyth** [comment]
Hi @vmsaipreeth -- sorry to hear you are having trouble.

We will be in a much better position to help you with these issues if you can provide reproducible examples.  

At the very least, we will require a representative example and the full traceback that is associated with that error.

**Reply by vmsaipreeth** [reply]
This is how we are trying to fetch data from an oracle catalog via trino

```
df_ibis = connection.sql(“”” SELECT col_a, col_b from ORACLE_CATALOG.SCHEMA.TABLE”””)
```
The table as many columns. However choosing only two columns for now.

DataTypes of the columns:
```
col_a decimal(15, 0) NOT NULL,
col_b varchar(7) NOT NULL
```

This throws an error while parsing the above SQL statement: 
```
ParseError: expected 'EOF' at 0:7
```

Please note that connection is established successfully and we are able to fetch data for hdfs and object storage catalogs. Observing the above issue only for oracle catalogs.

**Reply by cpcloud** [reply]
I suspect this is a type parsing issue on the ibis side.

**Reply by Tay0z** [reply]
@cpcloud is there a way to resolve this issue using a different query method?

**Reply by gforsyth** [reply]
@Tay0z -- if the Oracle database is available, you can try using the new Oracle backend to connect to it directly.  In any case, we'll attempt to fix the parsing issue.

**Comment by cpcloud** [comment, marked as answer]
This was a bug in parsing trino types. We hadn't implemented parsing for the all the types it supports.

Addressed in https://github.com/ibis-project/ibis/pull/6658

**Reply by vmsaipreeth** [reply]
Thanks @cpcloud for quick response. Highly appreciate it. 
Btw, when can we expect this feature to roll out?

**Reply by cpcloud** [reply]
I think we're going to release in a week or two, stay tuned!

**Reply by vmsaipreeth** [reply]
Awesome! Will keep an eye. Thank you!

**Comment by vmsaipreeth** [comment]
Hi there!

So, we upgraded ibis to ``` 6.1.0 ``` and had chance to play around. Thanks for the prompt response and fix.

We are now able to read oracle catalogs via trino 👍🏼.

Below code which uses ``` connection.sql()``` works well
```
connection = ibis.trino.connect( trino_connection_parameters)

df_ibis = connection.sql(“””SQL STATEMENT WITH ORACLE CATALOG VIA TRINO”””)
```

However, we noticed that the reading time as increased. Below are few metrics:

Time for reading Table A from object storage: 3.6 secs
Time for reading Table A from oracle catalog: 27.3 secs

Time for reading Table B from object storage: 1.8 secs
Time for reading Table B from oracle catalog: 16.3 secs

is there anyway that we can reduce the reading time for oracle catalogs that you would suggest?

Thanks in advance!

**Reply by cpcloud** [reply]
Let's discuss in an issue!
